### DISCUSSION TITLE
el-select odd behavior after build

### DISCUSSION THREAD

**Discussion by dmuy** [element-plus/element-plus#15652] [Q&A] [answered]
Hello,

I need help with regards to the `el-select` component being rendered differently after build. I'm using vue 3 + vite.

If you look at the screenshot below, production build (top) vs local build (bottom) el-select component is rendered differently. 
![image](https://github.com/element-plus/element-plus/assets/8830886/841069d5-1fdd-44ee-b171-a2c392f33457)

The production build also has `width: 11px` inline style on the select input element. I don't know what's causing this.
![image](https://github.com/element-plus/element-plus/assets/8830886/db525ec8-e2dc-45ee-80a4-6bccebe7baee)

**Comment by dmuy** [comment]
This basically causes UI issue like so:
![image](https://github.com/element-plus/element-plus/assets/8830886/69c4d5f1-3a45-4ee8-8f56-6e1c61bc2c16)
Production build (top) vs local build (bottom).

By "local build", I mean running `vite build` then hosting the `dist` folder on IIS (I'm on windows).

**Comment by chenxch** [comment]
try  `pnpm run dev --force`

**Reply by chenxch** [reply]
If it is not built for the same device, please check the version. The select component was refactored in version 2.5.0.

**Reply by dmuy** [reply]
Is running `pnpm run dev --force` helping for debugging it? Coz the problem is only when running `vite build`, not `vite dev`. Also I'm using `npm`.

**Reply by dmuy** [reply]
> If it is not built for the same device, please check the version. The select component was refactored in version 2.5.0.

I don't understand what you mean by "not built for the same device".

**Reply by dmuy** [reply]
For context, the "production build" I'm comparing it with is only a deployment (to a server via pipelines) with what I'm developing. Basically the same project, run on a server vs local machine after running `npm run build`

**Reply by chenxch** [reply]
Is the production build on the server? If so, you can check the version.

**Reply by dmuy** [reply]
Oh I get what you mean now, I deleted `node_modules` and the `package-lock.json` on my end and run `npm install` again

I got `"element-plus": "^2.4.4"` on `package.json` but resolves to version `2.5.3`. So, what changed? Do I have to specify the width now?

**Reply by chenxch** [reply, marked as answer]
Yes, see #15352 for information.
